google/cloud/storage/internal/sha256_hash.h

```diff
@@ -0,0 +1,40 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_SHA256_HASH_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_SHA256_HASH_H_
+
+#include "google/cloud/storage/version.h"
+#include <string>
+#include <vector>
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+namespace internal {
+
+/// Return the SHA256 hash (as raw bytes) of @p str.
+std::vector<std::uint8_t> Sha256Hash(std::string const& str);
+
+/// Return @p str encoded as a sequence of bytes.
+std::string HexEncode(std::vector<std::uint8_t> const& str);
```

Comment:
The argument name 'str' should probably be 'bytes'?

Reply:
Done.